src/Pact/Gas/Table.hs

```diff
@@ -275,6 +275,9 @@ tableGasModel gasConfig =
         GInterfaceDecl _interfaceName _iCode -> (_gasCostConfig_interfaceCost gasConfig)
         GModuleMemory i -> moduleMemoryCost i
         GPrincipal g -> fromIntegral g * _gasCostConfig_principalCost gasConfig
+        GMakeList2 len msz -> let
+          !glen = fromIntegral len
+          in glen + maybe 0 ((* glen) . intCost) msz
```

Comment:
Should this ever be 0 even if it's empty? Presumably the constructor still costs *something* to parse

Reply:
`msz` is purely for the `enumerate` case.

Reply:
Also, I don't think the bang pattern here helps. Here's how I read it:

1. Haskell will return a thunk for the whole binding.
2. When the thunk is forced, `glen` is forced first, due to the bang pattern.
3. Since the think will only be forced when the result is needed, `+` is also forced.
4. `+` is strict in its arguments, so `glen` would have been forced anyway.

If the desire was to enforce CSE, or only evaluating `glen` once, I think you get this just from the use of `let`. The bang doesn't change how much is evaluated, just the order.